src/main/java/chatBot/bot/Parser.java

```diff
@@ -0,0 +1,125 @@
+package chatBot.bot;
+
+import chatBot.command.*;
+import chatBot.exception.EmptyDescException;
+import chatBot.exception.InvalidCommandException;
+import chatBot.task.Deadline;
+import chatBot.task.Event;
+import chatBot.task.Task;
+import chatBot.task.ToDoTask;
+
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parser is a class to parse input Strings. It has 2 methods
+ *
+ * parseTask method is used exclusively for parsing tasks from a storage file,
+ * returns a Task to the caller.
+ *
+ * parse method is used to determine the command to go to next.
+ * returns a Command to the caller.
+ */
+
+public class Parser {
+    static Task parseTask(String s) {
+        String command = s.split(" ")[0];
+        String desc = s.substring(s.indexOf(" "));
+
+        desc = desc.stripLeading();
+
+        Task t = null;
+
+        if (command.equals("todo")) {
+            try {
+                t = new ToDoTask(desc);
+            } catch (Exception e) {
+                Ui.printAnything(e.getMessage());
+            }
+        } else if (command.equals("deadline")) {
+            try {
+                String[] arr = desc.split("/by");
+                try {
+                    t = new Deadline(arr[0].strip(), arr[1].strip());
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    Ui.printAnything("missing /by");
+                } catch (DateTimeParseException ex) {
+                    Ui.printAnything("Incorrect date format stored: " + arr[1].strip());
+                }
+            } catch (EmptyDescException e) {
+                Ui.printAnything(e.getMessage());
+            }
+        } else if (command.equals("event")) {
+            String[] arr = desc.split("/from");
+            String[] arr2 = new String[0];
+            try {
+                arr2 = arr[1].split("/to");
+            } catch (ArrayIndexOutOfBoundsException e) {
+                System.out.println("missing /from");
+            }
+            t = null;
+            try {
+                try {
+                    t = new Event(arr[0].strip(), arr2[0].strip(), arr2[1].strip());
+                } catch (ArrayIndexOutOfBoundsException ex) {
+                    System.out.println("missing /to");
+                } catch (DateTimeParseException ex) {
+                    System.out.println("Incorrect date format stored: " + arr[1].strip());
+                }
+            } catch (Exception e) {
+                Ui.printAnything(e.getMessage());
+            }
+        } else {
+            try {
+                throw new InvalidCommandException();
+            } catch (InvalidCommandException e) {
+                Ui.printAnything(e.getMessage());
+            }
+            //System.out.println("Unknown command: " + command);
+        }
+
+        if (t == null) {
+            System.out.println("null tasked returned back to Storage.load()");
+        }
+
+        return t;
+    }
+
+    static Command parse(String fullCommand) {
```

Comment:
Should these methods have header comments to make the parameters and return values more clear?